queries/elixir.scm

```diff
@@ -0,0 +1,41 @@
+(comment) @comment
+(anonymous_function) @anonymousFunction
+(arguments) @arguments_list
+(call
+  target: (_) @functionCallee) @functionCall @functionCallee.domain
+
+;;(keywords
+;;  (pair
+;;    key: (_) @collectionKey @collectionItem.trailing.end.startOf
+;;    value: (_) @value) @collectionItem @collectionItem.trailing.start.endOf
+;;    (",")? @collectionItem.trailing.start.startOf
+;;    (#allow-multiple! @collectionItem))
+
+;; How does surrounding whitespace work for this
+(map_content
+  (binary_operator
+      left: (_) @collectionKey @collectionKey.trailing.start.endOf
+      right: (_) @value @collectionKey.trailing.end.startOf) @collectionItem
+      ","? @collectionItem.trailing
+      (#shrink-to-match! @collectionItem "([^,]+)"))
+
+(call
+  target: (identifier) @ignore
+  (arguments
+    [
+      ; zero-arity functions with no parentheses
+      (identifier) @functionName
+      ; regular function clause
+      (call target: (identifier) @functionName)
+      ; function clause with a guard clause
+      (binary_operator
+        left: (call target: (identifier) @functionName)
+        operator: "when")
+    ])
+    (do_block
+      .
+      "do" @namedFunction.interior.start.endOf
+      .
+      "end" @namedFunction.interior.end.startOf
```

Comment:
worth having some "inside funk" tests for this one

Reply:
So this is mostly working at the moment (after removing that extra period), but I need commands like `chuck inside funk` to leave `end` on a separate line, or at least to leave a space between `do end`. Is that possible?

Reply:
Ideally
```
def elixir(foo) do
  bar = 5
end
```
- "chuck inside funk" ->
```
def elixir(foo) do
end
```

Right now it gives `def elixir(foo) doend`, which is a syntax error

Reply:
ooh huh interesting one 😅. as a hack, maybe we could shift the endpoint of the range to the left one?

Reply:
Interesting, is that something I'd do with a predicate?

Reply:
Yes, although I'm not sure the predicate exists yet, now that you mention it. You could maybe use the regex shrink predicate?